main.py

```diff
@@ -31,7 +31,11 @@ def generate_code_message(a, b):
 
 def send_code_message(update, context, is_anonymous):
     username = update.effective_user.username
-    name = f"{update.effective_user.first_name} {update.effective_user.last_name}".strip()
+    first_name = update.effective_user.first_name or ''
+    last_name = update.effective_user.last_name or ''
+    name = f"{first_name} {last_name}".strip()
+    is_anonymous = is_anonymouse or not username
```

Comment:
Typo here, and if user doesn't have a username, it's not necessarily anonymous.

e.g. they can have

`From: Christopher Goh` (without the telehandle)
instead of
`From: Anonymous`